code/asteroid/asteroid.cpp

```diff
@@ -1849,20 +1823,48 @@ static void asteroid_parse_section(asteroid_info *asip)
 		} else
 			Warning(LOCATION, "Invalid asteroid reference %i used for $Split in asteroids table, ignoring.", split_type);
 	}
+
+	if (optional_string("$Display Name:")) {
+		stuff_string(asip->display_name, F_NAME);
+	} else {
+		// take the first word of the name and append "debris"
+		SCP_string current_name = asip->name;
+		size_t split = current_name.find(' ');
+		asip->display_name = current_name.substr(0, split);
+		asip->display_name += " debris";
```

Comment:
Why not append " debris" to the end of the string?

Reply:
I did? I'm not sure what you mean

Reply:
In this code, you split the string at the first space.  A more accurate question would be, why are you splitting the string before appending?

Reply:
Oh, I'm stupid, I don't think I realized "debris" was the second word of the of ship asteroid names anyway. Still, the first two words is a bit more awkward of a default. Maybe strip numbers?

Reply:
What would a typical `asip->name` look like if you didn't modify it?

Reply:
'Terran Debris 1' 'Vasudan Debris 2' etc

Believe me, I would *really* prefer to just use the names as is. Maybe an explicit check for the retail ones? At least so everyone else isn't saddled with this weird default or forced to use $Display Name

Reply:
Oh, I see, it's actually specified in asteroids.tbl.

Why not use a `has_display_name` field like ship and weapon classes?

Reply:
I'm not sure how to go about it though, because it would have to work without any table changes for the retail ship asteroids, since the HUD was hardcoded to display them as "[species] debris", regardless of their actual name.

Reply:
Ah.  Then, yeah, do an explicit check, but I would do it this way: If the name is any species, followed by "debris", followed by a number, then the name is changed to "[species] debris".  Otherwise it is left alone.

Reply:
Should it be for *any* "[species] debris #" or just the specific retail names? I could hardcode those retail names in the code to be checked against. I'm likely going to need to do that anyway because I'm going to do another PR to check for the retail asteroids to suppress the inconsistent LOD number warnings they generate.

Reply:
It should be any species.  Remember that there are plenty of mods out there which defined new species, and some of them may have also defined species-specific debris.

Reply:
Fair enough.

Reply:
> What if it detects this style of naming and then uses the old index-based species "guess" for the species to fill in the new name

That sounds like the most sensible solution.